### DISCUSSION TITLE
Mission Influence tracking for BGS

### DISCUSSION THREAD

**Discussion by ACyprus** [EDCD/EDDI#2213] [EDDI VoiceAttack Q&A] [answered]
Hello team,

I'm looking for a way to track total Influence gained for a faction across a game session, when running BGS missions.

This is reported to the Squadron for BGS planning.

Currently, I write down the Influence reward following each mission hand-in, sum the total and pop it on Discord at the end of a night. This is cumbersome because I play in VR, and thus would love to be able to track this automatically with EDDI/VA without the need to remove my headset.

I can see the `((EDDI mission accepted))` event passes `{TXT:EDDI mission accepted influence}` but there doesn't seem to be an equivalent variable passed for `((EDDI mission completed))`.

Is there any way to access influence reward on mission hand-in, other than trying to correlate the two events above via corresponding `{DEC:EDDI mission accepted missionid}`?

Much thanks
o7

**Comment by Tkael** [comment, marked as answer]
It seems that at some point FDev added a `FactionEffects` property to the `MissionCompleted` event which we currently do not parse.
It should be possible to expose this data.
```json
{
    "timestamp": "2021-07-02T08:17:21Z",
    "event": "MissionCompleted",
    "Faction": "Noblemen of Nohock Ek",
    "Name": "Mission_AltruismCredits_name",
    "MissionID": 788930122,
    "Donation": "450000",
    "Donated": 450000,
    "FactionEffects": [{
            "Faction": "Noblemen of Nohock Ek",
            "Effects": [{
                    "Effect": "$MISSIONUTIL_Interaction_Summary_EP_up;",
                    "Effect_Localised": "The economic status of $#MinorFaction; has improved in the $#System; system.",
                    "Trend": "UpGood"
                }
            ],
            "Influence": [{
                    "SystemAddress": 5067658962353,
                    "Trend": "UpGood",
                    "Influence": "++"
                }
            ],
            "ReputationTrend": "UpGood",
            "Reputation": "++"
        }
    ]
}
```

Influence from the `MissionAccepted` event is not an adequate substitute, since the commander may or may not choose an influence reward after completing a mission.

**Reply by Tkael** [reply]
I've generated a ticket to investigate this further.

**Reply by ACyprus** [reply]
Thanks for checking this out, much appreciated. I would assume many CMDRs would find it useful.

For those interested, the issue to follow is #2214
